app/api/Files.scala

```diff
@@ -1623,6 +1616,26 @@ class Files @Inject()(
     }
   }
 
+  def bulkDeleteFiles() = PrivateServerAction (parse.json) {implicit request=>
+    request.user match {
+      case Some(user) => {
+        val fileIds = request.body.\("fileIds").asOpt[List[String]].getOrElse(List.empty[String])
+        if (fileIds.isEmpty){
+          BadRequest("No file ids supplied")
+        } else {
+          for (fileId <- fileIds){
```

Comment:
you can use a pattern like this to check many file permissions at once https://github.com/clowder-framework/clowder/blob/develop/app/util/SearchUtils.scala#L302